.azuredevops/components/AMDMIGraphX.yml

```diff
@@ -69,29 +75,29 @@ jobs:
       dependencySource: fixed
       fixedComponentName: half
       fixedPipelineIdentifier: $(half560-pipeline-id)
-# CI case: download latest default branch build
-  - ${{ if eq(parameters.checkoutRef, '') }}:
-    - template: ${{ variables.CI_TEMPLATE_PATH }}/steps/dependencies-rocm.yml
-      parameters:
-        dependencyList: ${{ parameters.rocmDependencies }}
+  - template: ${{ variables.CI_TEMPLATE_PATH }}/steps/dependencies-rocm.yml
+    parameters:
+      dependencyList: ${{ parameters.rocmDependencies }}
+      gpuTarget: $(JOB_GPU_TARGET)
+      # CI case: download latest default branch build
+      ${{ if eq(parameters.checkoutRef, '') }}:
         dependencySource: staging
-# manual build case: triggered by ROCm/ROCm repo
-  - ${{ if ne(parameters.checkoutRef, '') }}:
-    - template: ${{ variables.CI_TEMPLATE_PATH }}/steps/dependencies-rocm.yml
-      parameters:
-        dependencyList: ${{ parameters.rocmDependencies }}
+      # manual build case: triggered by ROCm/ROCm repo
+      ${{ elseif ne(parameters.checkoutRef, '') }}:
```

Comment:
Could we replace all these with an else statement?

Reply:
There is a request for amd-mainline ROCm stack for External CI in addition to amd-staging. so I will defer this type of change to a future PR when we have amd-mainline building automatically for the necessary components since that will need an elseif clause.